plutus-tx-constraints/src/Ledger/Tx/Constraints/OffChain.hs

```diff
@@ -321,16 +363,59 @@ processConstraint = \case
             <*> C.toCardanoTxOutValue vl
             <*> pure (toTxOutDatum md)
             <*> pure refScript
-
         unbalancedTx . tx . txOuts <>= [ out ]
 
     c -> error $ "Ledger.Tx.Constraints.OffChain: " ++ show c ++ " not implemented yet"
 
+{-
+-- | Add a typed input, checking the type of the output it spends. Return the value
+--   of the spent output.
+addOwnInput
```

Comment:
I thought I need to integrate this at first (when I was blocked because of the lack of balancing for cardano-tx), but it wasn't necessary, I'm deleting it.